lib/mix/lib/mix/tasks/profile.tprof.ex

```diff
@@ -0,0 +1,409 @@
+      :erlang.make_fun/3                                  1  7.69    2    2.00
+      Enum.each/2                                         1  7.69    2    2.00
+      anonymous fn/0 in :elixir_compiler_0.__FILE__/1     1  7.69    2    2.00
+      :erlang.integer_to_binary/1                         5 15.39    4    0.80
+      :erlang.apply/2                                     1 15.39    4    4.00
+      anonymous fn/3 in Enum.each/2                       5 19.23    5    1.00
+      String.Chars.Integer.to_string/1                    5 23.08    6    1.20
+
+      Profile done over 8 matching functions
+
+  The default output contains data gathered from all matching functions. The first
+  row after the header contains the sums of the partial results and the average time
+  for all the function calls listed. The following rows contain the function call,
+  followed by the number of times that the function was called, then by the percentage
+  of time that the call uses, then the total time for that function in microseconds,
+  and, finally, the average time per call in microseconds.
+
+  When `--matching` option is specified, call count tracing will be started only for
+  the functions matching the given pattern:
+
+      #                                               CALLS     % TIME µS/CALL
+      Total                                               5 100.0    6    1.20
+      String.Chars.Integer.to_string/1                    5 100.0    6    1.20
+
+      Profile done over 1 matching functions
+
+  The pattern can be a module name, such as `String` to count all calls to that module,
+  a call without arity, such as `String.split`, to count all calls to that function
+  regardless of arity, or a call with arity, such as `String.split/3`, to count all
+  calls to that exact module, function and arity.
+
+  ## Caveats
+
+  You should be aware that the code being profiled is running in an anonymous
+  function which is invoked by [`:tprof` module](https://www.erlang.org/doc/man/tprof.html).
+  Thus, you'll see some additional entries in your profile output. It is also
+  important to note that the profiler is stopped as soon as the code has finished running,
+  and this may need special attention, when: running asynchronous code as function calls which were
+  called before the profiler stopped will not be counted; running synchronous code as long
+  running computations and a profiler without a proper MFA trace pattern or filter may
+  lead to a result set which is difficult to comprehend.
+
+  You should expect a slowdown in your code execution using this tool since `:tprof` has
+  some performance impact on the execution, but the impact is considerably lower than
+  `Mix.Tasks.Profile.Fprof`. If you have a large system try to profile a limited
+  scenario or focus on the main modules or processes. Another alternative is to use
+  `Mix.Tasks.Profile.Cprof` that uses [`:cprof`](`:cprof`) and has a low performance degradation effect.
```

Comment:
```suggestion
  scenario or focus on the main modules or processes.
```

This is pretty much the same as cprof (which will be deprecated in the future anyway).

Reply:
Yes sorry as I mentioned in the description I should still rewrite the bulk of the moduledoc, it's still mostly `eprof`'s 😅